deep_qa/data/instances/mc_tuples_with_background_instance.py

```diff
@@ -0,0 +1,358 @@
+            self.get_lengths_from_indexed_tuples(self.background_indexed)
+        max_num_slots = max(max_num_slots, background_max_num_slots)
+
+        return {'num_options': len(self.answers_indexed),
+                'num_question_tuples': max_question_tuples,
+                'num_background_tuples': len(self.background_indexed),
+                'question_slot_length': max_answer_length,
+                'background_slot_length': background_max_slot_length,
+                'num_slots': max_num_slots}
+
+    # Helper method for self.get_lengths() -- gets the max lengths for a list of indexed tuples
+    @staticmethod
+    def get_lengths_from_indexed_tuples(indexed_tuples: List[List[List[int]]]):
+        max_num_slots = 0
+        max_slot_length = 0
+        for indexed_tuple in indexed_tuples:
+            num_slots = len(indexed_tuple)
+            max_num_slots = max(max_num_slots, num_slots)
+            for slot in indexed_tuple:
+                max_slot_length = max(max_slot_length, len(slot))
+        return max_num_slots, max_slot_length
+
+    @overrides
+    def pad(self, max_lengths: Dict[str, int]):
+        """
+        Pads (or truncates) all slots in the answers_indexed and background_indexed to specified lengths.
+        Additionally, adjusts the number of slots in all tuples to the desired number: if there are fewer
+        slots than desired, will add empty lists at the end, and if there are more will
+        concatenate object slots.
+        Also, pads the number of answer options, the number of tuples per answer option, and the number
+        of background tuples to the specified lengths, adding empty ones as needed.
+        Note: Adjusting the number of slots happens before adjusting the lengths of all the slots.
+        since truncating is done from the left, the earlier object slots ar more likely to be
+        lost.
+        """
+        desired_num_options = max_lengths['num_options']
+        desired_num_question_tuples = max_lengths['num_question_tuples']
+        desired_num_background_tuples = max_lengths['num_background_tuples']
+        desired_num_slots = max_lengths['num_slots']
+        desired_question_slot_length = max_lengths['question_slot_length']
+        desired_background_slot_length = max_lengths['background_slot_length']
+        # Pad the number of answers
+        num_options = len(self.answers_indexed)
+        if num_options > desired_num_options:
+            raise Exception("Case of too many answer options ({0} provided) isn't currently "
+                            "handled.".format(num_options))
+        elif num_options < desired_num_options:
+            # Add empty answer option
+            additional_options = [[] for _ in range(desired_num_options - num_options)]
+            self.answers_indexed += additional_options
+        # Pad the number of tuples per option
+        for curr_answer in range(len(self.answers_indexed)):
```

Comment:
Here you're using `curr_answer` to refer to an index, which is a confusing name to me.  I would call this `answer_index`, like you do below, or just do `for answer in self.answers_indexed:`, and append to or truncate `answer` directly.